transport/grpc/dial.go

```diff
@@ -129,7 +129,15 @@ func dial(ctx context.Context, insecure bool, o *internal.DialSettings) (*grpc.C
 	var grpcOpts []grpc.DialOption
 	if insecure {
 		grpcOpts = []grpc.DialOption{grpc.WithInsecure()}
-	} else if !o.NoAuth {
+	} else {
+		tlsConfig := &tls.Config{
+			GetClientCertificate: clientCertSource,
+		}
+		grpcOpts = []grpc.DialOption{
+			grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)),
+		}
+	}
+	if !o.NoAuth {
```

Comment:
One new behavior that would come of this refactor is that users could now accidentally send credentials over an insecure (unencrypted) connection. Should we force `o.NoAuth = true` when `insecure == true`? This would retain "backwards compatible" behavior for those using `WithoutAuthentication` but not `WithInsecure`. Not sure if these even worked in the first place because I get errors trying to use that configuration.

Reply:
we _could_ just do `if !o.NoAuth && !insecure {` to retain the exact previous behavior.

Reply:
I think I like that idea. Can drop a todo in or file an issue to see if we should be more lenient in the future.

Reply:
Ok, I've added this!